scripts/migrate_dry_run.py

```
@@ -0,0 +1,154 @@
+import sqlalchemy as sa
+
+
+_TABLES = ("nodes", "memories", "edges", "paths", "search_documents", "glossary_keywords")
+
+
+async def _row_counts(db) -> dict[str, int]:
+    counts: dict[str, int] = {}
+    async with db.session() as s:
+        for table in _TABLES:
+            try:
+                result = await s.execute(sa.text(f"SELECT COUNT(*) FROM {table}"))
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)